internal/http/services/owncloud/ocdav/put.go

```diff
@@ -154,9 +154,20 @@ func (s *svc) handlePut(ctx context.Context, w http.ResponseWriter, r *http.Requ
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
+	opaqueMap := map[string]*typespb.OpaqueEntry{}
```

Comment:
I would suggest to change this into
```suggestion
	var opaque *typespb.Opaque
```

Then use `utils.AppendPlainToOpaque` to add `HeaderOCMTime` and `HeaderUploadLength`. This way the opaque magic is hidden in `utils` package